src/scene/light.js

```diff
@@ -317,54 +362,16 @@ class Light {
         this._updateFinalColor();
     }
 
-    get shadowMap() {
-
-        if (this._shadowCamera) {
-            const rt = this._shadowCamera.renderTarget;
-            if (this._type === LIGHTTYPE_OMNI) {
-                return rt.colorBuffer;
-            }
-
-            return this._isPcf && this.device.webgl2 ? rt.depthBuffer : rt.colorBuffer;
-        }
-
-        return null;
-    }
-
-    _destroyShadowMap() {
-        if (this._shadowCamera) {
-            if (!this._isCachedShadowMap) {
-                var rt = this._shadowCamera.renderTarget;
-                var i;
-                if (rt) {
-                    if (rt.length) {
-                        for (i = 0; i < rt.length; i++) {
-                            rt[i].destroyTextureBuffers();
-                            rt[i].destroy();
-                        }
-                    } else {
-                        rt.destroyTextureBuffers();
-                        rt.destroy();
-                    }
-                }
-            }
-            this._shadowCamera.renderTarget = null;
-            this._shadowCamera = null;
-            this._shadowCubeMap = null;
-            if (this.shadowUpdateMode === SHADOWUPDATE_NONE) {
-                this.shadowUpdateMode = SHADOWUPDATE_THISFRAME;
-            }
-        }
-    }
-
     updateShadow() {
         if (this.shadowUpdateMode !== SHADOWUPDATE_REALTIME) {
             this.shadowUpdateMode = SHADOWUPDATE_THISFRAME;
         }
     }
 
     layersDirty() {
-        this._scene.layers._dirtyLights = true;
+        if (this?._scene?.layers) {
```

Comment:
I don't think you need the first optional chaining operator here since `this` should be valid.

Reply:
cheers, agreed and done.